lib/em/pure_ruby.rb

```diff
@@ -228,10 +292,47 @@ def set_pending_connect_timeout sig, tm
   end
 end
 
-
 module EventMachine
   # @private
   class Error < Exception; end
+  # @private
+  class ConnectionError < RuntimeError; end
+  # @private
+  class ConnectionNotBound < RuntimeError; end
+  # @private
+  class UnknownTimerFired < RuntimeError; end
+  # @private
+  class Unsupported < RuntimeError; end
+end
```

Comment:
Let's consider this PR ahead of my other one -- just a heads up that I may scoop these out into a separate lib/em/error.rb as a followup step.